semanticdb/semanticdb3/semanticdb3.md

```diff
@@ -141,129 +187,100 @@ It represents a location inside a document, such as a line inside a text file.
+e.g. coming from multiple different versions of Scala - then all references
+to those definitions will have `SymbolOccurrence.symbol` equal to the same
+`_root_.scala.Int#` symbol, and SemanticDB will not be able to provide
+information to distinguish these references from each other.
 
 In the future, we may extend SemanticDB to allow for multiple definitions
 that under current rules would correspond to the same global symbol.
-In the meanwhile, when global uniqueness is required, tool authors are advised to
-accompany global symbols with `SymbolInformation.location`.
+In the meanwhile, when global uniqueness is required, tool authors are advised
+to accompany global symbols with `SymbolInformation.location`.
 
 **Local symbols**. Correspond to a definition that isn't global (see above).
-Local symbol format is defined by the following two rules:
-  * Local symbols must start with `local`, so that they can be easily
-    distinguished from global symbols.
-  * Local symbols must be unique within the underlying document.
 
-For example, `x` in `def identity[T](x: T): T` may be modelled by local symbols
-`local0`, `local_x`, `local_identity_x`, as long as these names are unique within
-the underlying document. The same logic applies to the type parameter `T`,
-which is also a local definition.
+Local symbol format is language-agnostic and is a concatenation of `local`
+and a decimal number. For example, `x` in a Scala method
+`def identity[T](x: T): T` may be modelled by local symbols `local0`, `local1`,
+etc. The same logic applies to the type parameter `T`, which is also a local
+definition.
 
 Local symbols must be unique within the underlying document, but they don't have
 to be unique across multiple documents. For example, at the time of writing
-`semanticdb-scalac` produces local symbols named `local0`, `local1`, etc,
-with the counter resetting to zero for every new document.
+the Scalac-based SemanticDB producer generates local symbols named `local0`,
+`local1`, etc, with the counter resetting to zero for every new document.
 
 In the future, we may extend SemanticDB to make local symbols more unique,
 but we haven't yet found a way to do that without sacrificing performance
 and payload size. In the meanwhile, when global uniqueness is required,
 tool authors are advised to accompany local symbols with `TextDocument.uri`.
 
-**Multi symbols**. Are used to model references to a set of multiple definitions at once.
-This is occasionally useful to support corner cases of Scala, e.g. identifiers
-in imports that can refer to both a class and an object with the same name,
-or references to unresolved overloaded methods.
+**Multi symbols**. Represent references to multiple definitions at once.
+For example, this is occasionally useful to support corner cases of Scala,
+e.g. identifiers in imports that can refer to both a class and an object with
+the same name, or references to unresolved overloaded methods.
```

Comment:
overloaded methods is a better motivating example than "corner case of scala"

Reply:
This formulation was around like that for a long time. Unresolved overloaded methods is a corner case though, I think. You don't see them in typechecked Scala programs.